keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepository.kt

```diff
@@ -1,18 +1,23 @@
 package com.netflix.spinnaker.keel.persistence
 
-import com.netflix.frigga.ami.AppVersion
-import com.netflix.rocket.semver.shaded.DebianVersionComparator
 import com.netflix.spinnaker.keel.api.ArtifactStatus
 import com.netflix.spinnaker.keel.api.ArtifactType
 import com.netflix.spinnaker.keel.api.DeliveryArtifact
 import com.netflix.spinnaker.keel.api.DeliveryConfig
-import org.slf4j.LoggerFactory
-import org.springframework.util.comparator.NullSafeComparator
 
 interface ArtifactRepository {
 
+  /**
+   * Creates or updates a registered artifact
+   */
+  // todo eb: allow the same artifact with different versioning strategies to be registered
+  // so that two users aren't battling
+  // maybe require an artifact to be associated with a delivery config?
+  // maybe have an artifact also have a name to reference?
   fun register(artifact: DeliveryArtifact)
```

Comment:
^ This is the big registration of artifacts change. I think we need artifacts to have one more piece of information to tie them to a delivery config so that we can support two people caring about the same image but choosing different versioning strategies. That's a change for another PR, though.